src/compiler/crystal/interpreter/compiler.cr

```diff
@@ -3327,4 +3350,12 @@ class Crystal::Repl::Compiler < Crystal::Visitor
 
   private macro nop
   end
+
+  private def with_node_override(node_override : ASTNode)
+    old_node_override = @node_override
+    @node_override = node_override
+    value = yield
+    @node_override = old_node_override
+    value
```

Comment:
Perhaps an additional safeguard?

```suggestion
    yield
  ensure
    @node_override = old_node_override
```

Reply:
Thanks but I don't want the performance overhead of an ensure.

Reply:
(not sure if there is any, but there's also no point in doing an ensure because if some thing fails the compiler isn't used anymore /